src/Database/Type/DateTimeType.php

```diff
@@ -165,7 +169,7 @@ public function marshal($value)
                 $date = new $class($value);
                 $compare = true;
             }
-            if ($compare && $date && $date->format($this->_format) !== $value) {
+            if ($compare && $date && !$this->_compare($date, $value)) {
```

Comment:
This check was originally intended to ensure that an ambiguous input format wasn't used resulting in a different datetime value than was in the request data. Perhaps that isn't useful, an we should instead rely on validation and app logic.